src/neo-vm/ExecutionEngine.cs

```diff
@@ -585,7 +585,32 @@ private void ExecuteOp(OpCode opcode, ExecutionContext context)
                     case OpCode.CHECKSIG:
                         {
                             byte[] pubkey = context.EvaluationStack.Pop().GetByteArray();
-                            byte[] signature = context.EvaluationStack.Pop().GetByteArray();
+                            StackItem item = context.EvaluationStack.Pop();
+                            byte[] signature = item.GetByteArray();
+                            if (item is VMArray array1)
+                            {
+                                byte[][] signatures = array1.Select(p => p.GetByteArray()).ToArray();
+                                if (signatures.Length != 1)
+                                {
+                                    State |= VMState.FAULT;
+                                    return;
+                                }
+                                signature = signatures[0];
+                            }
+                            // preparing CHECKSIG for Neo 3.0 format: expects "verify" (0x766572696679)
+                            if(signature.Equals(new byte[]{0x76,0x65,0x72,0x69,0x66,0x79}) && context.EvaluationStack.Length > 0)
```

Comment:
1. This `Equals` will always return `false`.
1. This code will not be merged into master branch because the proposal to unify the entry point itself has not been accepted.

Reply:
Ok, I'll remove this part.